### PR TITLE
Ra dspdc 1715 notification improvements

### PR DESCRIPTION
**Pull request by raaidbroad**
## Why
It would help us and our stakeholders to have better visibility into whether or not a new release was processed and whether or not a new snapshot was published. Likewise, it'd be helpful to have the step and error message of a failure. Given the difficulty of threading information through/bubbling information up, we will start by simply adding the environment to success/failure messages, and adding simple success observability.
[Relevant ticket](https://broadinstitute.atlassian.net/browse/DSPDC-1715)

## This PR
- Adds exit handlers to send Slack messages for "A new release file is being ingested" and "A new snapshot is being published"
- Adds the environment to the success/failure messages

### CODE REVIEW

File: orchestration/templates/ingest-xml-archive.yaml [hunk 1]
```diff
@@ -367,3 +368,28 @@ spec:
             json='{"release_date":"{{ $releaseDate }}","archive_path":"{{ $archiveId }}"}'
             echo "$json" > archive.json
             gsutil cp archive.json {{ $uploadPath }}
+
+    ##
+    ## Send a Slack notification describing the final status of the workflow.
+    ##
+    - name: send-slack-notification
```
Comment (raaidbroad) [inline orchestration/templates/ingest-xml-archive.yaml:375]: Yes, this is copy pasted. No, it is not DRY. Keeping Argo yaml DRY is a time sink, so if anyone feels strongly about this, I can make it DRY, but be warned that it'll take longer than it should.

File: orchestration/templates/process-and-reingest-release.yaml [hunk 2]
```diff
@@ -459,3 +460,28 @@ spec:
     ## Inject template used to poll TDR jobs.
     {{- include "argo.poll-ingest-job" . | indent 4 }}
     {{- end }}
+
+      ##
+      ## Send a Slack notification describing the final status of the workflow.
+      ##
+    - name: send-slack-notification
```
Comment (raaidbroad) [inline orchestration/templates/process-and-reingest-release.yaml:467]: Same DRY comment as above

File: orchestration/templates/cron-workflow.yaml [hunk 3]
```diff
@@ -64,5 +66,5 @@ spec:
             - -H
             - 'Content-type: application/json'
             - -d
-            - '{"text": "Workflow $(WORKFLOW_ID) entered state: $(WORKFLOW_STATE)"}'
+            - '{"text": "Workflow $(WORKFLOW_ID) in env *$(WORKFLOW_ENV)* entered state: $(WORKFLOW_STATE)"}'
```
Comment (aherbst-broad) [inline orchestration/templates/cron-workflow.yaml:69]: 🤔 Might be good to *bold* the WORKFLOW_STATE as well, for readability?
```suggestion
            - '{"text": "Workflow $(WORKFLOW_ID) in env *$(WORKFLOW_ENV)* entered state: *$(WORKFLOW_STATE)*"}'
```

File: orchestration/templates/ingest-xml-archive.yaml (hunk 1 above)
Comment (aherbst-broad) [reply orchestration/templates/ingest-xml-archive.yaml:375]: We have a [template](https://github.com/broadinstitute/monster-helm/blob/master/charts/argo-templates/templates/notification.yaml) for this in monster-helm already.  For some reason I thought we had switched over to using it, but evidently not. Might be worth spending 10 minutes and seeing if it'll work ok here? Not a blocker for this PR though.

File: orchestration/templates/process-and-reingest-release.yaml [hunk 4]
```diff
@@ -459,3 +460,28 @@ spec:
...
+              secretKeyRef:
+                name: {{ .Values.notification.slackUrl.secretName }}
+                key: {{ .Values.notification.slackUrl.secretKey }}
+        command: [curl]
+        args:
+        - -XPOST
+        - -H
+        - 'Content-type: application/json'
+        - -d
+        - '{"text": "Workflow $(WORKFLOW_ID) in env *$(WORKFLOW_ENV)* is cutting a new snapshot."}'
```
Comment (aherbst-broad) [inline orchestration/templates/process-and-reingest-release.yaml:486]: ❓ It seems like this fires _after_ the snapshot is cut, given my reading of the `publish-processing-results` template. If that's the case, a minor rewording is in order, to get the tense right:
```suggestion
        - '{"text": "Workflow $(WORKFLOW_ID) in env *$(WORKFLOW_ENV)* has cut a new snapshot."}'
```

**Review by aherbst-broad — APPROVED** [review]

**Review by eparkmonster — APPROVED** [review]

**Review by quazi-broad — APPROVED** [review]
Looks good to me aside from the DRY comments you already noted

**Pull request merged**